### PR TITLE
Add --extensive flag to list command

### PR DESCRIPTION
**Pull request by guibog**
I did a plugin for this, but I think it may make more sense as a flag in list command, maybe.

BTW travis ci failed on my other PR #1353,  but `python setup.py test` is all green on my env. So I expect this PR to break ci, but not sure what I can do to fix it.

### CODE REVIEW

File: beets/ui/commands.py [hunk 1]
```diff
@@ -969,11 +969,41 @@ def list_items(lib, query, album, fmt=''):
...
 def list_func(lib, opts, args):
-    list_items(lib, decargs(args), opts.album)
+    if opts.extensive:
+        list_items_extensive(lib, decargs(args), opts.album)
+    else:
+        list_items(lib, decargs(args), opts.album)
 
 
 list_cmd = ui.Subcommand('list', help='query the library', aliases=('ls',))
+list_cmd.parser.add_option('-e', '--extensive', action='store_true', help='list all attributes')
```
Comment (tomjaspers) [inline beets/ui/commands.py]: Line is too long (96 > 79 characters).

You can run `tox -e flake8` for styling, or just `tox` to test everything, on your env. That way it'll match Travis.

**Comment by tomjaspers** [comment]
The other reason why Travis is failing on both PRs (except the line being too long on this one), is because the docs generator has been updated and the beets configuration has not.

**Comment by brunal** [comment]
Hi,
I have trouble understanding the goal of this PR. Isn't it what the `info` plugin does? Otherwise can't you get that functionality with the `--format` flag?

**Comment by brunal** [comment]
@tomjaspers I just fixed the docs config in commit  e71caded81.

**Comment by guibog** [comment]
@brunal I did not know about info plugin. It seems I can get the same functionality with `beet info -l`, with a slightly different layout (left padded instead of right padded). I don't think `--format` would work because I do not know in advance the names of the attributes, or is there some special magic in it?

Also, I still think that it could be convenient for users to expand the list command, instead of requiring them to check all plugins and enable info. But clearly info is better than my option, as it allows summarizing the attributes.

**Comment by sampsyo** [comment]
We also talked about merging the functionality of `list` and `info` in #1287. I don't currently quite see how that could work without making both more complicated to understand. Proposals welcome.

**Comment by sampsyo** [comment]
We also talked about merging the functionality of `list` and `info` in #1287. I don't currently quite see how that could work without making both more complicated to understand. Proposals welcome.

**Comment by guibog** [comment]
@sampsyo the `list` command is not extremely complex right now, is it? Maybe less then 30 lines of code as of now. Adding a flag listing an extensive list of attributes can be useful for people taking care of their media files (which I understand is the target of beets project), and did not look like very difficult when I tried in this PR. I would even add a more explicit display showing if an attribute is or is not "user defined" (in `items_attribute` table) and if it is saved in the file or not.

However I would not add what is requested in issue #1287 because filtering output is better done with `grep` or format, and displaying tags of files not in beets library is better done with `exiftool -v4`.

Then there is the `summary` option in info which is useful but maybe a bit more complex to add to `list` command. It could work a bit like a "report" option, which accumulates attribute values and display some stats at the end.

**Comment by sampsyo** [comment]
No, it's not complex—that's what we like about it. :smiley:

I'm still not 100% clear on what we gain from duplicating `info`'s functionality inside the `list` command. Is it just that the command is available by default?

**Comment by guibog** [comment]
> what we gain from duplicating info's functionality inside the list command. 

I would not duplicate, but move info's functionality in the list command. Yes, having it available by default is nice, but also list and info are very similar. When I found out about list, I thought: this is what I need to use to display attributes of some songs. Then I tried to display all of them, but did not think that I should look for this functionality in another command.

**Comment by sampsyo** [comment]
Aha, thanks for clarifying! Perhaps this is a documentation issue more than an implementation one—is there anything we can do to make it clearer when a user should look for `info`?

The reason I'm resisting merging these is because the functionality is actually quite orthogonal from a CLI perspective. If we added a `--details` flag to `info`, it would create weird conflicts where the other flags (`-f`, for example) became irrelevant and a different set of flags (selecting fields, summarization, etc.) would suddenly be enabled. Better to keep these as separate commands to avoid hidden dependencies.

There's also one fundamental difference at the moment: `list` works only on items (and albums) in the library; `info` can also work directly on files (it was originally conceived as a replacement for `mp3info` and the like). This would need to be resolved first.

**Comment by guibog** [comment]
Hmm, I see. In fact allowing `info` command to work also on files makes it much more orthogonal to `'list` command than I thought. Also, it may make it a bit unclear. For instance what happens if I have a file named 'foo' in cwd and some tracks matching "foo" in my lib? The "query" of info command becomes something different.

By the way, I just found a bug in `info`: if I remove the original file (e.g. unplug the usb drive) I get a traceback. Will try to fix or send a bug.

So it could be the case that it's actually info plugin that need some improvements and can be, later, enabled by default. Or, maybe, another plugin, e.g. "inspect", could be created to give extensive information about an item (e.g. which attributes are in the file and in lib, which only on one side, type of attributes, etc.) and if it works well it could subsume current info plugin's functionality. What do you think?

**Comment by sampsyo** [comment]
Sure! Enabling `info` by default could indeed be a good idea. It doesn't require any additional dependencies or anything, so it won't cost much to put it in core.

I'd lean toward incrementally improving the current command over starting a new one from scratch. Seems like a useful project.

I'll close this ticket for now, but feel free to continue the discussion here.

**Comment by tomjaspers** [comment]
@guibog I've taken care of the traceback on IOError in `info`, thanks for reporting it !

**Comment by sampsyo** [comment]
Thanks, @tomjaspers!

**Pull request closed without merging**